icinga-app/icinga2.cmake

```diff
@@ -1,29 +1,46 @@
 #!/bin/sh
 ICINGA2_BIN=@CMAKE_INSTALL_FULL_LIBDIR@/icinga2/sbin/icinga2
 
-if test "x`uname -s`" = "xLinux" -a "x$1" = "xconsole"; then
-  libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
-  if test $? -eq 0; then
-    libedit_path=`echo $libedit_line | cut -f3 -d' '`
-    if test -n "$libedit_path"; then
-      libdir=`dirname -- $libedit_path`
-      found=0
-      for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
-        found=1
-        break
-      done
-      if test $found -eq 0; then
-        libdir2=/`echo $libdir | cut -f3- -d/`
-        for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+if test "x`uname -s`" = "xLinux"; then
+  if test "x$1" = "xconsole"; then
+    libedit_line=`ldd $ICINGA2_BIN 2>&1 | grep libedit`
+    if test $? -eq 0; then
+      libedit_path=`echo $libedit_line | cut -f3 -d' '`
+      if test -n "$libedit_path"; then
+        libdir=`dirname -- $libedit_path`
+        found=0
+        for libreadline_path in `ls -1r -- $libdir/libreadline.so.* 2>/dev/null`; do
           found=1
           break
         done
-      fi
-      if test $found -gt 0; then
-        export LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        if test $found -eq 0; then
+          libdir2=/`echo $libdir | cut -f3- -d/`
+          for libreadline_path in `ls -1r -- $libdir2/libreadline.so.* 2>/dev/null`; do
+            found=1
+            break
+          done
+        fi
+        if test $found -gt 0; then
+          LD_PRELOAD="$libreadline_path:$LD_PRELOAD"
+        fi
       fi
     fi
   fi
+
+  for lib in /usr/lib{,64,/{{i386,x86_64}-linux-gnu,arm-linux-gnueabihf}}/libjemalloc.so.{1,2}; do
+    if test -e "$lib"; then
+      # Assumptions:
+      # "$ICINGA2_BIN" is present and dynamically linked and --version is a no-op.
+      # If anything is wrong with LD_PRELOAD="$lib", LD will complain on stderr. E.g.:
+      # ERROR: ld.so: object '/usr/lib/i386-linux-gnu/libjemalloc.so.2' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
+      if LD_PRELOAD="$lib" "$ICINGA2_BIN" --version 2>&1 1>/dev/null |grep -qFwe LD_PRELOAD
+      then true
+      else LD_PRELOAD="${lib}:$LD_PRELOAD"
```

Comment:
Since there's no `break` in here, this would write multiple `.so` files into `LD_PRELOAD`.